zio-http/shared/src/main/scala/zio/http/endpoint/openapi/JsonSchema.scala

```diff
@@ -318,129 +318,141 @@ object JsonSchema {
       case SegmentCodec.Combined(_, _, _) => throw new IllegalArgumentException("Combined segment is not supported.")
     }
 
-  def fromZSchemaMulti(schema: Schema[_], refType: SchemaStyle = SchemaStyle.Inline): JsonSchemas = {
+  def fromZSchemaMulti(
+    schema: Schema[_],
+    refType: SchemaStyle = SchemaStyle.Inline,
+    seen: Set[Predef.String] = Set.empty,
+  ): JsonSchemas = {
     val ref = nominal(schema, refType)
-    schema match {
-      case enum0: Schema.Enum[_] if enum0.cases.forall(_.schema.isInstanceOf[CaseClass0[_]]) =>
-        JsonSchemas(fromZSchema(enum0, SchemaStyle.Inline), ref, Map.empty)
-      case enum0: Schema.Enum[_]                                                             =>
-        JsonSchemas(
-          fromZSchema(enum0, SchemaStyle.Inline),
-          ref,
-          enum0.cases
-            .filterNot(_.annotations.exists(_.isInstanceOf[transientCase]))
-            .flatMap { c =>
-              val key    =
-                nominal(c.schema, refType)
-                  .orElse(nominal(c.schema, SchemaStyle.Compact))
+    if (ref.exists(seen.contains)) {
+      JsonSchemas(RefSchema(ref.get), ref, Map.empty)
+    } else {
+      val seenAndRef = seen ++ ref
```

Comment:
The name sounds like a tuple. seeUpdated? 🙂

Reply:
Renamed to `seenWithCurrent`, so that it is clear why it has been updated.